satpy/readers/tropomi_l2.py

```diff
@@ -0,0 +1,131 @@
+
+            # some other file handler knows how to load this
+            if is_avail is not None:
+                yield is_avail, ds_info
+
+            var_name = ds_info.get('file_key', ds_info['name'])
+            # logger.debug("Evaluating previously configured variable: %s", var_name)
+            matches = self.file_type_matches(ds_info['file_type'])
+            # we can confidently say that we can provide this dataset and can
+            # provide more info
+            if matches and var_name in self:
+                logger.debug("Handling previously configured variable: %s", var_name)
+                handled_variables.add(var_name)
+                new_info = ds_info.copy()  # don't mess up the above yielded
+                yield True, new_info
+            elif is_avail is None:
+                # if we didn't know how to handle this dataset and no one else did
+                # then we should keep it going down the chain
+                yield is_avail, ds_info
+
+        # This is where we dynamically add new datasets
+        # We will sift through all groups and variables, looking for data matching
+        # the geolocation bounds
+
+        # Iterate over dataset contents
+        for var_name, val in self.file_content.items():
+            # Only evaluate variables
+            if isinstance(val, netCDF4.Variable):
+                logger.debug("Evaluating new variable: %s", var_name)
+                var_shape = self[var_name + "/shape"]
+                logger.debug("Dims:{}".format(var_shape))
+                if (var_shape == lat_shape):
+                    logger.debug("Found valid additional dataset: %s", var_name)
+                    # Skip anything we have already configured
+                    if (var_name in handled_variables):
+                        logger.debug("Already handled, skipping: %s", var_name)
+                        continue
+                    handled_variables.add(var_name)
+                    last_index_separator = var_name.rindex('/')
+                    last_index_separator = last_index_separator + 1
+                    var_name_no_path = var_name[last_index_separator:]
+                    logger.debug("Using short name of: %s", var_name_no_path)
+                    # Create new ds_info object
+                    new_info = {
+                        'name': var_name_no_path,
+                        'file_key': var_name,
+                        'coordinates': ['longitude', 'latitude'],
+                        'file_type': self.filetype_info['file_type'],
+                        'resolution': None,
+                    }
+                    yield True, new_info
+
+    def get_metadata(self, data, ds_info):
+        metadata = {}
+        metadata.update(data.attrs)
+        metadata.update(ds_info)
+        metadata.update({
+            'platform_shortname': self.platform_shortname,
+            'start_time': self.start_time,
+            'end_time': self.end_time,
```

Comment:
It doesn't look like you have `sensor` defined, could you add that? Do the files have `units` and `standard_name` defined?